Assignment2-Ancestors/Q1.py

```diff
@@ -0,0 +1,44 @@
+from Tree import BinaryTree
+
+def find_ancestors(tree, key):
+    """
+    Find ancestors of the first node with the given value while preorder traversal and save the Nodes in a list that return.
+    :param tree: Tree
+    :param key: any
+    :return: list(Nodes)
+    """
+    node = tree.find(key)
+    if node:

```

Comment:
About last_common_element() function. Yes, there was that silly mistake with indexes and even some others, but funny thing is that results while testing were ok... It was just lucky cases,  or unlucky as I've posted bad solution :)
But I've fixed it and now it is suppose to work good.